### ISSUE TITLE
[JENKINS-62215] antisamy-markup-formatter-plugin v2.0 filters input fields from uno-choice plugin

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/active-choices-plugin#757]
<p>With Active Choices and OWASP Markup Formatter Plugin v2.0 installed, "<tt>input</tt>" elements are being stripped from "Formatted HTML" parameters.  This is not an issue with OWASP Markup Formatter Plugin v1.8.  Since this is one of the main use-cases of the Active Choices plugin, I assume this is a bug.  There also doesn't seem to be a way to configure the markup formatter that the Active Choices plugin uses.</p>

<p> </p>

<p>Attached are screenshots of the parameter config, the working "<tt>input</tt>" field with v1.8, and the missing input field with v2.0.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/apottere.jpg" title="apottere's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere">apottere</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215" target="_blank">antisamy-markup-formatter-plugin v2.0 filters input fields from uno-choice plugin</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kinow.png" title="kinow's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kinow">kinow</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: active-choices-plugin, antisamy-markup-formatter-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-07-08T21:32:42+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-12-08
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>With Active Choices and OWASP Markup Formatter Plugin v2.0 installed, "<tt>input</tt>" elements are being stripped from "Formatted HTML" parameters.  This is not an issue with OWASP Markup Formatter Plugin v1.8.  Since this is one of the main use-cases of the Active Choices plugin, I assume this is a bug.  There also doesn't seem to be a way to configure the markup formatter that the Active Choices plugin uses.</p>

<p> </p>

<p>Attached are screenshots of the parameter config, the working "<tt>input</tt>" field with v1.8, and the missing input field with v2.0.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins 2.219<br/>
Active Choices (uno-choice) v2.0<br/>
OWASP Markup Formatter Plugin (antisamy-markup-formatter-plugin) v1.8/v2.0
```
</details>
<details><summary><i>6 attachments</i></summary>

- [bruno-screenshot-1.png](https://issues.jenkins.io/secure/attachment/51775/bruno-screenshot-1.png)
  > ![bruno-screenshot-1.png](https://issues.jenkins.io/secure/attachment/51775/bruno-screenshot-1.png)
- [bruno-screenshot-2.png](https://issues.jenkins.io/secure/attachment/51776/bruno-screenshot-2.png)
  > ![bruno-screenshot-2.png](https://issues.jenkins.io/secure/attachment/51776/bruno-screenshot-2.png)
- [bruno-screenshot-3.png](https://issues.jenkins.io/secure/attachment/51777/bruno-screenshot-3.png)
  > ![bruno-screenshot-3.png](https://issues.jenkins.io/secure/attachment/51777/bruno-screenshot-3.png)
- [markup-formatter-1.8.png](https://issues.jenkins.io/secure/attachment/51154/markup-formatter-1.8.png)
  > ![markup-formatter-1.8.png](https://issues.jenkins.io/secure/attachment/51154/markup-formatter-1.8.png)
- [markup-formatter-2.0.png](https://issues.jenkins.io/secure/attachment/51153/markup-formatter-2.0.png)
  > ![markup-formatter-2.0.png](https://issues.jenkins.io/secure/attachment/51153/markup-formatter-2.0.png)
- [parameter-config.png](https://issues.jenkins.io/secure/attachment/51155/parameter-config.png)
  > ![parameter-config.png](https://issues.jenkins.io/secure/attachment/51155/parameter-config.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kinow.png" title="kinow's avatar" /> kinow:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=390340&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-390340">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for the report and for the taking your time to provide detailed description and screenshots.</p>

<p>We were preparing a release for the next days (waiting till pending pull request gets reviewed/tested by other maintainers). I will try to include some time to take a look and try to reproduce/fix this one.</p>

<p>Bruno</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for the report and for the taking your time to provide detailed description and screenshots.</p>

<p>We were preparing a release for the next days (waiting till pending pull request gets reviewed/tested by other maintainers). I will try to include some time to take a look and try to reproduce/fix this one.</p>

<p>Bruno</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ioannis.png" title="ioannis's avatar" /> ioannis:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=390489&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-390489">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a> first I would upgrade Active Choices to the latest release.</p>

<p> </p>

<p>I have used antisamy-markup-formatter v2.0 successfully with Active Choices v2.2.3 on Jenkins LTS v2.222.1</p>

<p>Please, make sure that in Jenkins 'Configure Global Security' the Markup Formatter option is set to 'Safe HTML' Hope this helps.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a> first I would upgrade Active Choices to the latest release.</p>

<p> </p>

<p>I have used antisamy-markup-formatter v2.0 successfully with Active Choices v2.2.3 on Jenkins LTS v2.222.1</p>

<p>Please, make sure that in Jenkins 'Configure Global Security' the Markup Formatter option is set to 'Safe HTML' Hope this helps.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/apottere.jpg" title="apottere's avatar" /> apottere:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=390498&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-390498">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ioannis" class="user-hover" rel="ioannis">ioannis</a> The issue occurs whether the <tt>Markup Formatter</tt> option is set to <tt>Plain Text</tt> or <tt>Safe HTML</tt>.</p>

<p> </p>

<p>I wasn't able to find version 2.2.3 of the Active choices plugin, either in GitHub Releases (<a href="https://github.com/jenkinsci/active-choices-plugin/releases" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/active-choices-plugin/releases)</a> or the update site (<a href="https://updates.jenkins.io/download/plugins/uno-choice/" class="external-link" target="_blank" rel="nofollow noopener">https://updates.jenkins.io/download/plugins/uno-choice/)</a>.  From the README (<a href="https://github.com/jenkinsci/active-choices-plugin#version-23-2020" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/active-choices-plugin#version-23-2020</a>) it looks like 2.3 is the next pending version.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ioannis" class="user-hover" rel="ioannis">ioannis</a> The issue occurs whether the <tt>Markup Formatter</tt> option is set to <tt>Plain Text</tt> or <tt>Safe HTML</tt>.</p>

<p> </p>

<p>I wasn't able to find version 2.2.3 of the Active choices plugin, either in GitHub Releases (<a href="https://github.com/jenkinsci/active-choices-plugin/releases" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/active-choices-plugin/releases)</a> or the update site (<a href="https://updates.jenkins.io/download/plugins/uno-choice/" class="external-link" target="_blank" rel="nofollow noopener">https://updates.jenkins.io/download/plugins/uno-choice/)</a>.  From the README (<a href="https://github.com/jenkinsci/active-choices-plugin#version-23-2020" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/active-choices-plugin#version-23-2020</a>) it looks like 2.3 is the next pending version.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ioannis.png" title="ioannis's avatar" /> ioannis:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=390502&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-390502">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a> It's possible that I have an intermediate AC snapshot as I usually do some testing before final release. Nonetheless, I seriously doubt that the issue will turn out to be active choices itself. You seem to have isolated the issue to the antisamy-markup-formatter-plugin v2.0, but all I'm saying is that this version works for me and my (rather permissive Jenkins instance)</p>

<p>I use the following Jenkins startup parameters that you may also want to test. I suspect that the CustomJellyContext may be of special interest here</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">-Dhudson.model.DirectoryBrowserSupport.CSP= -Dorg.kohsuke.stapler.jelly.CustomJellyContext.escapeByDefault=<span class="code-quote">"<span class="code-keyword">false</span>"</span></pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a> It's possible that I have an intermediate AC snapshot as I usually do some testing before final release. Nonetheless, I seriously doubt that the issue will turn out to be active choices itself. You seem to have isolated the issue to the antisamy-markup-formatter-plugin v2.0, but all I'm saying is that this version works for me and my (rather permissive Jenkins instance)</p>

<p>I use the following Jenkins startup parameters that you may also want to test. I suspect that the CustomJellyContext may be of special interest here</p>

<pre>
-Dhudson.model.DirectoryBrowserSupport.CSP= -Dorg.kohsuke.stapler.jelly.CustomJellyContext.escapeByDefault=<span class="code-quote">"<span class="code-keyword">false</span>"</span></pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/apottere.jpg" title="apottere's avatar" /> apottere:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=392453&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-392453">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ioannis" class="user-hover" rel="ioannis">ioannis</a> sorry, must have missed the email for your comment.  I created a git repo with a run script, the current versions of both plugins, and the java opts you specify.  When I run it, the input field is not displayed.</p>

<p> </p>

<p><a href="https://github.com/apottere/JENKINS-62215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apottere/JENKINS-62215</a></p>

<p> </p>

<p>I also included ".bak" versions of both plugins, which are the older (working) versions.  Let me know if you get different results.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ioannis" class="user-hover" rel="ioannis">ioannis</a> sorry, must have missed the email for your comment.  I created a git repo with a run script, the current versions of both plugins, and the java opts you specify.  When I run it, the input field is not displayed.</p>

<p> </p>

<p><a href="https://github.com/apottere/<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/62215">JENKINS-62215</a>" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apottere/<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/62215">JENKINS-62215</a></a></p>

<p> </p>

<p>I also included ".bak" versions of both plugins, which are the older (working) versions.  Let me know if you get different results.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/apottere.jpg" title="apottere's avatar" /> apottere:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=393417&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-393417">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It looks like this was an intentional change in the markup-formatter plugin: <a href="https://github.com/jenkinsci/antisamy-markup-formatter-plugin/pull/12" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/antisamy-markup-formatter-plugin/pull/12</a></p>



<p>It really seems like there's no way you should be able to see `input` fields with any version of the Active Choices plugin after that markup formatter update.  Either this issue should be fixed or the docs for Active Choices should be updated to make it clear that using form fields is no longer an option with the groovy sandbox (and the corresponding sanitation).</p>
</pre>
</details>

</ul>
</details>

<p>It looks like this was an intentional change in the markup-formatter plugin: <a href="https://github.com/jenkinsci/antisamy-markup-formatter-plugin/pull/12" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/antisamy-markup-formatter-plugin/pull/12</a></p>



<p>It really seems like there's no way you should be able to see `input` fields with any version of the Active Choices plugin after that markup formatter update.  Either this issue should be fixed or the docs for Active Choices should be updated to make it clear that using form fields is no longer an option with the groovy sandbox (and the corresponding sanitation).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kinow.png" title="kinow's avatar" /> kinow:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=393441&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-393441">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That's funny <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a>, I'm using the `master` branch of the git repo, and `mvn clean install` then `mvn hpi:run`. Trying to reproduce the issue before switching the branch to your PR, but no luck.</p>

<p> <span class="image-wrap" style=""><a id="51775_thumb" href="https://issues.jenkins.io/secure/attachment/51775/51775_bruno-screenshot-1.png" title="bruno-screenshot-1.png" file-preview-type="image" file-preview-id="51775" file-preview-title="bruno-screenshot-1.png"><img src="https://issues.jenkins.io/secure/thumbnail/51775/_thumb_51775.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>But no matter what I try, it still renders the input fine for me?</p>

<p> <span class="image-wrap" style=""><a id="51776_thumb" href="https://issues.jenkins.io/secure/attachment/51776/51776_bruno-screenshot-2.png" title="bruno-screenshot-2.png" file-preview-type="image" file-preview-id="51776" file-preview-title="bruno-screenshot-2.png"><img src="https://issues.jenkins.io/secure/thumbnail/51776/_thumb_51776.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p> <span class="image-wrap" style=""><a id="51777_thumb" href="https://issues.jenkins.io/secure/attachment/51777/51777_bruno-screenshot-3.png" title="bruno-screenshot-3.png" file-preview-type="image" file-preview-id="51777" file-preview-title="bruno-screenshot-3.png"><img src="https://issues.jenkins.io/secure/thumbnail/51777/_thumb_51777.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>Any idea what I could be missing? The plugin integration was a contribution after we were blocked by Jenkins security, and it "just worked", so I never bothered digging deeper to understand how everything works. So I may be missing some important step?</p>

<p>Thanks
Bruno</p>
</pre>
</details>

</ul>
</details>

<p>That's funny <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a>, I'm using the `master` branch of the git repo, and `mvn clean install` then `mvn hpi:run`. Trying to reproduce the issue before switching the branch to your PR, but no luck.</p>

<p> <span class="image-wrap" style=""><a id="51775_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/51775/bruno-screenshot-1.png" title="bruno-screenshot-1.png" file-preview-type="image" file-preview-id="51775" file-preview-title="bruno-screenshot-1.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/51775/bruno-screenshot-1.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>But no matter what I try, it still renders the input fine for me?</p>

<p> <span class="image-wrap" style=""><a id="51776_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/51776/bruno-screenshot-2.png" title="bruno-screenshot-2.png" file-preview-type="image" file-preview-id="51776" file-preview-title="bruno-screenshot-2.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/51776/bruno-screenshot-2.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p> <span class="image-wrap" style=""><a id="51777_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/51777/bruno-screenshot-3.png" title="bruno-screenshot-3.png" file-preview-type="image" file-preview-id="51777" file-preview-title="bruno-screenshot-3.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/51777/bruno-screenshot-3.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p>Any idea what I could be missing? The plugin integration was a contribution after we were blocked by Jenkins security, and it "just worked", so I never bothered digging deeper to understand how everything works. So I may be missing some important step?</p>

<p>Thanks<br/>
Bruno</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/apottere.jpg" title="apottere's avatar" /> apottere:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=393523&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-393523">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kinow" class="user-hover" rel="kinow">kinow</a> can you try running the following in your script console?</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">import</span> jenkins.model.Jenkins
<span class="code-keyword">import</span> hudson.markup.RawHtmlMarkupFormatter

println(<span class="code-quote">"Markup Formatter Version: "</span> + Jenkins.get().getPlugin(<span class="code-quote">"antisamy-markup-formatter"</span>).getWrapper().getVersion())
println(<span class="code-quote">"Formatted: <span class="code-quote">'${RawHtmlMarkupFormatter.INSTANCE.translate('</span>&lt;input type="</span>text<span class="code-quote">" name="</span>value<span class="code-quote">" value="</span>bar<span class="code-quote">" /&gt;<span class="code-quote">')}'</span>"</span>)
</pre>
</div></div>
<p> </p>

<p>That gives the following result on my test instance:</p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Markup Formatter Version: 2.0
Formatted: ''
</pre>
</div></div>
<p>And when I downgrade the plugins:</p>

<p> </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Markup Formatter Version: 1.8
Formatted: <span class="code-quote">'&lt;input type=<span class="code-quote">"text"</span> name=<span class="code-quote">"value"</span> value=<span class="code-quote">"bar"</span>&gt;'</span>
</pre>
</div></div>
<p> </p>

<p>AFAICT it looks like formatting with RawHtmlMarkupFormatter is unavoidable when using the groovy sandbox, per this line: <a href="https://github.com/biouno/uno-choice-plugin/blob/master/src/main/java/org/biouno/unochoice/model/GroovyScript.java#L174" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/biouno/uno-choice-plugin/blob/master/src/main/java/org/biouno/unochoice/model/GroovyScript.java#L174</a></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kinow" class="user-hover" rel="kinow">kinow</a> can you try running the following in your script console?</p>

<p> </p>

<pre>
<span class="code-keyword">import</span> jenkins.model.Jenkins
<span class="code-keyword">import</span> hudson.markup.RawHtmlMarkupFormatter

println(<span class="code-quote">"Markup Formatter Version: "</span> + Jenkins.get().getPlugin(<span class="code-quote">"antisamy-markup-formatter"</span>).getWrapper().getVersion())
println(<span class="code-quote">"Formatted: <span class="code-quote">'${RawHtmlMarkupFormatter.INSTANCE.translate('</span><input type="</span>text<span class="code-quote">" name="</span>value<span class="code-quote">" value="</span>bar<span class="code-quote">" /><span class="code-quote">')}'</span>"</span>)
</pre>
<p> </p>

<p>That gives the following result on my test instance:</p>

<p> </p>

<pre>
Markup Formatter Version: 2.0
Formatted: ''
</pre>
<p>And when I downgrade the plugins:</p>

<p> </p>

<p> </p>

<pre>
Markup Formatter Version: 1.8
Formatted: <span class="code-quote">'<input type=<span class="code-quote">"text"</span> name=<span class="code-quote">"value"</span> value=<span class="code-quote">"bar"</span>>'</span>
</pre>
<p> </p>

<p>AFAICT it looks like formatting with RawHtmlMarkupFormatter is unavoidable when using the groovy sandbox, per this line: <a href="https://github.com/biouno/uno-choice-plugin/blob/master/src/main/java/org/biouno/unochoice/model/GroovyScript.java#L174" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/biouno/uno-choice-plugin/blob/master/src/main/java/org/biouno/unochoice/model/GroovyScript.java#L174</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kinow.png" title="kinow's avatar" /> kinow:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=393542&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-393542">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That's interesting. I got </p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Result

Markup Formatter Version: 2.0
Formatted: ''
</pre>
</div></div>


<p>I tried setting up security, without luck. I still couldn't reproduce the issue. I tried to upgrade the Jenkins version in pom.xml to the latest LTS version. The new UI is neat, but broke active choices (I think I saw that coming from another issue about replacing tablesby divs).</p>

<p>Then I ran out of time for testing it sorry. Next time I have some spare time to work on the plugin I will try running the LTS war with the latest version instead.</p>

<p>Things that I am concerned when reviewing the PR:</p>

<p>1. need to properly reproduce the issue before merging and releasing it
2. would be good to get someone  - if possible - from Jenkins security team to confirm the plugin won't be blocked. I think the markup plugin was added due to a CVE with active choices that removed the plugin from update center. That's something that we need to try to avoid.
3. update docs about it</p>

<p>Cheers
Bruno</p>
</pre>
</details>

</ul>
</details>

<p>That's interesting. I got </p>



```
Result

Markup Formatter Version: 2.0
Formatted: ''

```


<p>I tried setting up security, without luck. I still couldn't reproduce the issue. I tried to upgrade the Jenkins version in pom.xml to the latest LTS version. The new UI is neat, but broke active choices (I think I saw that coming from another issue about replacing tablesby divs).</p>

<p>Then I ran out of time for testing it sorry. Next time I have some spare time to work on the plugin I will try running the LTS war with the latest version instead.</p>

<p>Things that I am concerned when reviewing the PR:</p>

<p>1. need to properly reproduce the issue before merging and releasing it<br/>
2. would be good to get someone  - if possible - from Jenkins security team to confirm the plugin won't be blocked. I think the markup plugin was added due to a CVE with active choices that removed the plugin from update center. That's something that we need to try to avoid.<br/>
3. update docs about it</p>

<p>Cheers<br/>
Bruno</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/apottere.jpg" title="apottere's avatar" /> apottere:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=393586&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-393586">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kinow" class="user-hover" rel="kinow">kinow</a> it looks like you got the right output in the script console, are you sure the job was using the groovy sandbox?  It only uses the markup sanitization when the groovy sandbox is used.  I'm not sure if you saw but I have a repo that uses the jenkins docker image to reproduce the issue: <a href="https://github.com/apottere/JENKINS-62215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apottere/JENKINS-62215</a></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kinow" class="user-hover" rel="kinow">kinow</a> it looks like you got the right output in the script console, are you sure the job was using the groovy sandbox?  It only uses the markup sanitization when the groovy sandbox is used.  I'm not sure if you saw but I have a repo that uses the jenkins docker image to reproduce the issue: <a href="https://github.com/apottere/<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/62215">JENKINS-62215</a>" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apottere/<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/62215">JENKINS-62215</a></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kinow.png" title="kinow's avatar" /> kinow:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=394021&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394021">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Fixed in 2.4. Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a></p>
</pre>
</details>

</ul>
</details>

<p>Fixed in 2.4. Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=apottere" class="user-hover" rel="apottere">apottere</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kinow.png" title="kinow's avatar" /> kinow:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62215?focusedId=405207&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-405207">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>In 2.4</p>
</pre>
</details>

</ul>
</details>

<p>In 2.4</p>
